tsconfig.json

```diff
@@ -0,0 +1,20 @@
+{
```

Comment:
I would suggest adding the following options:

#### `forceConsistentCasingInFileNames: true`
> Disallow inconsistently-cased references to the same file.

This would cause `tsc` to fail if you tried to import a file named `foobar` as `import foobar from './fooBar';`

#### `esModuleInterop: true`
> Emit __importStar and __importDefault helpers for runtime babel ecosystem compatibility and enable --allowSyntheticDefaultImports for typesystem compatibility.

The above description isn't very helpful. In short, it makes `commonJS` modules work as if they were `es6` modules. [Here's](https://stackoverflow.com/a/56348146/983194) a good SO answer on the problem it solves.

#### `noImplicitReturns: true`
> Report an error when not all code paths in function return a value.

#### `strict: true`
> Enable all strict type checking options.
Enabling --strict enables --noImplicitAny, --noImplicitThis, --alwaysStrict, --strictBindCallApply, --strictNullChecks, --strictFunctionTypes and --strictPropertyInitialization.

This is a must, IMO. Although some of these might be more trouble then they're worth at this stage of the TS adoption. If that's the case, I suggest enabling `strict` and then disabling the explicit rules that this repo isn't ready to enable yet. Of all the flags that this enables, I think `alwaysStrict`, `strictNullChecks`, and `strictFunctionTypes` are the most important. That said, trying to enable them all is worthwhile.

Reply:
we are not ready for these full strict setting yet.
`strictNullChecks` will also cause a lot of issues for us as we adopt TS.
for now 
`forceConsistentCasingInFileNames`, `alwaysStrict`, and `strictFunctionTypes` should be OK.